src/lib/pubkey/curve448/curve448_utils/curve448_gf.h

```diff
@@ -0,0 +1,141 @@
+/*
+* Curve448 Gf Modulo 2^448 - 2^224 - 1
+* (C) 2024 Jack Lloyd
+*     2024 Fabian Albert - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+#ifndef BOTAN_CURVE_448_GF_H_
+#define BOTAN_CURVE_448_GF_H_
+
+#include <botan/mem_ops.h>
+#include <botan/types.h>
+
+#include <array>
+#include <span>
+
+namespace Botan {
+
+/**
+ * This class represents a GF element in the field GF(2^448 - 2^224 - 1). Computations are
+ * performed using optimized operations as defined in the paper:
+ * "Reduction Modulo 2^448 - 2^224 - 1" by Kaushik Nath and Palash Sarkar
+ * (https://eprint.iacr.org/2019/1304).
+ *
+ * The representation of the field element is a 448-bit uint, stored in little-endian order
+ * as 7*64bit words. Note that the internal representation is not necessarily canonical, i.e.
+ * the value might be larger than the prime modulus. When calling the to_bytes() method, the
+ * canonical representation is returned.
+ */
+class Gf448Elem {
```

Comment:
Perhaps its worthwhile defining some commonly used types:
   ```C++
   using byte_span = std::span<const uint8_t, to_bytes(448)>;
   using word_span = std::span<const uint64_t, to_words(448)>;
   // same for arrays, perhaps
   ```

Reply:
I don't use such aliases because they hide the actual type from the user. Also, I would need one variant with and without const. However, I think using constants for 56 (bytes per 448) and 7 (words per 448) are sensible.